waspc/cli/src/Wasp/Cli/Command/Test.hs

```diff
@@ -0,0 +1,49 @@
+module Wasp.Cli.Command.Test
+  ( test,
+  )
+where
+
+import Control.Concurrent.Async (race)
+import Control.Concurrent.MVar (newMVar)
+import Control.Monad.Except (throwError)
+import Control.Monad.IO.Class (liftIO)
+import StrongPath (Abs, Dir, (</>))
+import StrongPath.Types (Path')
+import Wasp.Cli.Command (Command, CommandError (..))
+import Wasp.Cli.Command.Common (findWaspProjectRootDirFromCwd)
+import Wasp.Cli.Command.Compile (compile)
+import Wasp.Cli.Command.Message (cliSendMessageC)
+import Wasp.Cli.Command.Watch (watch)
+import qualified Wasp.Cli.Common as Common
+import qualified Wasp.Generator
+import Wasp.Generator.Common (ProjectRootDir)
+import qualified Wasp.Message as Msg
+
+test :: [String] -> Command ()
+test [] = throwError $ CommandError "Not enough arguments." "Expected: wasp test client <args>"
+test ("client" : args) = watchAndTest $ Wasp.Generator.testWebApp args
+test ("server" : _args) = throwError $ CommandError "Invalid arguments." "Server testing not yet implemented."
+test _ = throwError $ CommandError "Invalid arguments." "Expected: wasp test client <args>"
+
+watchAndTest :: (Path' Abs (Dir ProjectRootDir) -> IO (Either String ())) -> Command ()
+watchAndTest testRunner = do
+  waspRoot <- findWaspProjectRootDirFromCwd
+  let outDir = waspRoot </> Common.dotWaspDirInWaspProjectDir </> Common.generatedCodeDirInDotWaspDir
+
+  cliSendMessageC $ Msg.Start "Starting compilation and setup phase. Hold tight..."
```

Comment:
Since the output is long:
```
🐝 --- Starting compilation and setup phase. Hold tight... ------------------------


🐝 --- Compiling wasp project... --------------------------------------------------


🐝 --- Setting up database... -----------------------------------------------------


✅ --- Database successfully set up. ----------------------------------------------


✅ --- Your wasp project has successfully compiled. -------------------------------


🐝 --- Listening for file changes... ----------------------------------------------


🐝 --- Running tests... -----------------------------------------------------------
```

Maybe consider removing this message and merging the two from below into `Watching for file changes and running tests ...`